### PR TITLE
(SDI-2483) Auto-Load plugins via temp similar to cli

### LINKED ISSUE #1502: Do not execute binaries directly from the auto-load path
Currently, plugins loaded by the snaptel cli via 'snaptel plugin load file_name' are copied to a temporary directory before they are executed. Plugins placed in the auto-load directory are loaded directly from that filepath without going through this process. This prevents plugin files from getting updated when Snap is running since they are in use. We would like snap plugins to behave the same whether they are loaded via the autoloader or snaptel plugin load cli.

### PR DESCRIPTION
**Pull request by rashmigottipati**
Fixes #1502 

Summary of changes:
- Load the plugins via auto-loader using the temp path similar to running snaptel plugin load cli
- Removed the autoload field

Testing done:
- Manual
- Legacy tests

@intelsdi-x/snap-maintainers

### CODE REVIEW

File: core/plugin.go [hunk 1]
```diff
@@ -95,21 +99,56 @@ type SubscribedPlugin interface {
...
-	}
-	err := rp.generateCheckSum()
-	if err != nil {
-		return nil, err
+	path      string
+	checkSum  [sha256.Size]byte
+	signature []byte
+}
+
+func NewRequestedPlugin(path, tmp string, b []byte) (*RequestedPlugin, error) {
```
Comment (IRCody) [inline core/plugin.go]: Can we add a comment about the expected behavior of this? Especially with b being nil vs not being nil.

File: pkg/fileutils/file.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"os"
+	"path/filepath"
+	"runtime"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// WriteFile creates a temporary directory for loading plugins.
+// Plugins loaded by the snaptel cli and plugins placed in the auto-load directory go through this route of copying the plugin binaries to the temp dir and executing from temp.
+// This function takes the name of the original file, path to the original file and the content of the file, and copies it to a a temp file after creating one.
```
Comment (jcooklin) [inline pkg/fileutils/file.go]: Include a fragment that states what it returns specifically.

File: core/plugin.go [hunk 3]
```diff
@@ -95,21 +99,60 @@ type SubscribedPlugin interface {
...
-	}
-	err := rp.generateCheckSum()
-	if err != nil {
-		return nil, err
+	path      string
+	checkSum  [sha256.Size]byte
+	signature []byte
+}
+
+// NewRequestedPlugin returns a Requested Plugin which represents the plugin path, check sum and signature
```
Comment (jcooklin) [inline core/plugin.go]: Provide details about the expected arguments (what are path, tmp, b? can they be empty or nil, etc)

File: core/plugin.go [hunk 4]
```diff
@@ -95,21 +99,60 @@ type SubscribedPlugin interface {
...
+// Both REST API and snaptel cli would call this before loading a plugin.
+func NewRequestedPlugin(path, tmp string, b []byte) (*RequestedPlugin, error) {
```
Comment (jcooklin) [inline core/plugin.go]: name or fileName would be more meaningful than tmp

**Comment by jcooklin** [comment]
LGTM

**Pull request merged**